File: QE-Python_selenium/qe-selenium-python/Activity11.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/dynamic-controls")
print(driver.title)

wait = WebDriverWait(driver, 15)

toggle_checkbox_btn = (By.XPATH, "//button[contains(.,'Toggle Checkbox')]")
checkbox = (By.CSS_SELECTOR, "input[type='checkbox']")

# Click to remove checkbox
wait.until(EC.element_to_be_clickable(toggle_checkbox_btn)).click()

# Wait until checkbox is invisible/removed
wait.until(EC.invisibility_of_element_located(checkbox))
print("Checkbox is removed (not visible).")

# Click to add checkbox back
wait.until(EC.element_to_be_clickable(toggle_checkbox_btn)).click()

# Wait for checkbox to be present again
cb = wait.until(EC.presence_of_element_located(checkbox))

# Wait until checkbox is clickable
wait.until(EC.element_to_be_clickable(checkbox))
print("Checkbox appeared.")

# Select checkbox if not already selected
if not cb.is_selected():
    cb.click()

print("Checkbox selected:", cb.is_selected())

driver.quit()

File: QE-Java-Selenium/QE-selenium-java/javaselenium/src/main/java/javaselenium/demos/Activity11.java
package javaselenium.demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity11 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/dynamic-controls");
        System.out.println(driver.getTitle());

        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));

        By toggleCheckboxBtn = By.xpath("//button[contains(.,'Toggle Checkbox')]");
        By checkbox = By.cssSelector("input[type='checkbox']");   
        wait.until(ExpectedConditions.elementToBeClickable(toggleCheckboxBtn)).click(); 
        wait.until(ExpectedConditions.invisibilityOfElementLocated(checkbox));
        System.out.println("Checkbox is removed (not visible).");
 
        wait.until(ExpectedConditions.elementToBeClickable(toggleCheckboxBtn)).click(); 
        WebElement cb = wait.until(ExpectedConditions.presenceOfElementLocated(checkbox));
        wait.until(ExpectedConditions.elementToBeClickable(cb));
        System.out.println("Checkbox appeared.");
 
        if (!cb.isSelected()) {
            cb.click();
        }
        System.out.println("Checkbox selected: " + cb.isSelected()); 
        driver.quit();
    }
}